apps/backend/src/database/migrations/1686030966770-file-ingestion-migration.ts

```diff
@@ -0,0 +1,77 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class migration1686030966770 implements MigrationInterface {
+  name = 'file-ingestion-migration1686030966770';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE TABLE "file_ingestion_rules" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "program" character varying NOT NULL, "cash_cheques_filename" character varying, "pos_filename" character varying, "transactions_filename" character varying, "retries" integer NOT NULL DEFAULT '0', CONSTRAINT "PK_092531b74dee2a92a21d2e1d981" PRIMARY KEY ("id"))`
+    );
+    await queryRunner.query(
+      `CREATE TABLE "program_daily_upload" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "created_at" TIMESTAMP NOT NULL DEFAULT now(), "daily_date" date NOT NULL, "success" boolean NOT NULL, "retries" integer NOT NULL, "ruleId" uuid, CONSTRAINT "PK_24196c3558e5e3bf7617a2b453d" PRIMARY KEY ("id"))`
+    );
+    await queryRunner.query(
+      `CREATE TABLE "file_uploaded" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "created_at" TIMESTAMP NOT NULL DEFAULT now(), "source_file_type" character varying NOT NULL DEFAULT 'TDI17', "source_file_name" character varying NOT NULL, "source_file_length" integer NOT NULL, "daily_upload_id" uuid, CONSTRAINT "PK_27908e2def3d89bd889a9cd65dd" PRIMARY KEY ("id"))`
```

Comment:
I am curious what the benefit is to defaulting to TDI17 here?

Reply:
It's a good point, it's more me following the existing structure where TDI17 is the default for source file name

Reply:
Hmm okay, maybe we chuck this in the general error handling for input data bucket. Seems like it introduces some data quality risk for no real benefit (IMO).